Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Collections/BaseCollection.cs

```diff
@@ -0,0 +1,134 @@
+                {
+                    emptyNodes.Add(NodeList[i]);
+                }
+            }
+
+            // Now delete the empty nodes
+            for (int i = 0; i < emptyNodes.Count; i++)
+            {
+                NodeList.Remove(emptyNodes[i]);
+            }
+
+            emptyNodes.Clear();
+
+            // Check when children change and adjust
+            for (int i = 0; i < this.transform.childCount; i++)
+            {
+                Transform child = this.transform.GetChild(i);
+
+                if (!ContainsNode(child) && (child.gameObject.activeSelf || !IgnoreInactiveTransforms))
+                {
+                    CollectionNode node = new CollectionNode();
+
+                    node.Name = child.name;
+                    node.transform = child;
+                    NodeList.Add(node);
+                }
+            }
+
+            switch (SortType)
+            {
+                case SortTypeEnum.None:
+                    break;
+
+                case SortTypeEnum.Transform:
+                    NodeList.Sort(delegate (CollectionNode c1, CollectionNode c2) { return c1.transform.GetSiblingIndex().CompareTo(c2.transform.GetSiblingIndex()); });
+                    break;
+
+                case SortTypeEnum.Alphabetical:
+                    NodeList.Sort(delegate (CollectionNode c1, CollectionNode c2) { return c1.Name.CompareTo(c2.Name); });
+                    break;
+
+                case SortTypeEnum.AlphabeticalReversed:
+                    NodeList.Sort(delegate (CollectionNode c1, CollectionNode c2) { return c1.Name.CompareTo(c2.Name); });
+                    NodeList.Reverse();
+                    break;
+
+                case SortTypeEnum.TransformReversed:
+                    NodeList.Sort(delegate (CollectionNode c1, CollectionNode c2) { return c1.transform.GetSiblingIndex().CompareTo(c2.transform.GetSiblingIndex()); });
+                    NodeList.Reverse();
+                    break;
+            }
+
+            LayoutChildren();
+
+            if (OnCollectionUpdated != null)
+            {
+                OnCollectionUpdated.Invoke(this);
+            }
+        }
+
+        /// <summary>
+        /// Internal function to check if a node exists in the NodeList.
+        /// </summary>
+        /// <param name="node">A <see cref="Transform"/> of the node to see if it's in the NodeList</param>
+        /// <returns></returns>
+        private bool ContainsNode(Transform node)
+        {
+            for (int i = 0; i < NodeList.Count; i++)
+            {
+                if (NodeList[i] != null)
+                {
+                    if (NodeList[i].transform == node)
+                    {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Protected Internal function for laying out all children when UpdateCollection is called.
```

Comment:
Similar to the earlier comment. No need to specify the visibility in the comments